### PR TITLE
Document generation module

### PR DESCRIPTION
**Pull request by Rafalkufel**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+test:
+	pytest .
+
+type=specjalne
+generate:
+	python app/create_documents.py --type ${type}
+
+style:
+	black .
+	isort .
```
Comment (w1stler) [inline Makefile:10]: You can use `isort` with default `black`'s settings: `isort --profile black .`

File: app/docx/generate_document.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                document.write(filename=file_source_path, arcname=file_arch_path)
+
+            document.write(filename=self.footnotes_file_path, arcname=os.path.join("word", "footnotes.xml"))
+            document.writestr(data=self.get_rendered_document(), zinfo_or_arcname=os.path.join("word", "document.xml"))
+
+
+class Documents:
+    def __init__(self, documents: list[DocumentsTypes], document_data: dict, destination_path: str):
+        """
+        params:
```
Comment (w1stler) [inline app/docx/generate_document.py:60]: IMO iff you're adding typing in def, there's no need to specify params.

File: app/docx/schemas.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    address_first_parent_checkbox: bool = False
+    issue: Issues
+    issue_short: str | None
+    period: str
+    reason: Reasons
+    second_reason: Reasons | None
+    reason_genetive: str | None
+    application_description_genetive: str | None
+    multiple_disability: list[str] = []
+    no: str
```
Comment (w1stler) [inline app/docx/schemas.py:83]: `no` as a string?

**Review by w1stler — COMMENTED** [review]
Some comments after a quick peek into the code. ;-)

File: app/docx/schemas.py (hunk 3 above)
Comment (Rafalkufel) [reply app/docx/schemas.py:83]: eg `007/2022`

File: app/docx/schemas.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+
+class DocumentData(BaseModel):
+    # raw
+    child: ChildData
+    address_child_checkbox: bool = False
+    address_first_parent_checkbox: bool = False
+    issue: Issue
+    period: str
+    reasons: list[Reason] = Field(None, description="Can be one or moe reasons")
```
Comment (w1stler) [inline app/docx/schemas.py:85]: more - typo

File: app/docx/schemas.py [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+            )
+        return values
+
+    @root_validator
+    def check_multiple_disability(cls, values):
+        """Check excluded together disabilities was selected"""
+        reasons = values["reasons"]
+        if Reason.UNIEMOZLIWIAJACY.value in reasons and Reason.ZNACZNIE_UTRUDNIAJACY.value in reasons:
+            raise ValueError(f"Reasons: {', '.join(reasons)} can't be issued together.")
+        if intelecual_reasons := [reason for reason in reasons if reason in Reason.intelectual_reasons()]:
```
Comment (w1stler) [inline app/docx/schemas.py:133]: probably we could only check >1

File: app/docx/schemas.py [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+            if len(intelecual_reasons) > 1:
+                raise ValueError(f"Two intelecutal reasons: {', '.join(intelecual_reasons)} can't be issued together.")
+        if Reason.GLEBOKIE.value in reasons and len(reasons) > 1:
+            raise ValueError("Profound intelectual disability can't be coupled.")
+        if any([reason for reason in reasons if reason in Reason.social_maladjustment_reasons()]) and len(reasons) > 1:
+            raise ValueError("Social maladjustment reasons can be coupled with any other reason.")
```
Comment (w1stler) [inline app/docx/schemas.py:139]: typo - `can't be`

File: app/docx/schemas.py [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+        return values
+
+    @root_validator
+    def calculate_reason(cls, values):
+        if len(values["reasons"]) > 1:
```
Comment (w1stler) [inline app/docx/schemas.py:144]: probably could be oneliner

File: app/docx/schemas.py [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+        values["full_name"] = f"{values['first_name']} {values['last_name']}"
+
+        return values
+
+
+class ChildData(PersonalData):
+    pesel: str
+    birth_date: date
+    birth_place: str
+    klass: str | None
```
Comment (w1stler) [inline app/docx/schemas.py:42]: `_class` or `use alias`? Or a comment maybe?

**Review by w1stler — APPROVED** [review]

**Pull request merged**